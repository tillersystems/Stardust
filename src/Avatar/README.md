# Avatar

### Usage

```jsx
import Avatar from 'components/Avatar';

// or

import { Avatar } from 'components';
```

<!-- STORY -->

### Properties

- `className` - className needed by styled component.
- `name` - user name.
- `size` - size of the avatar.
- `src` - source image of a user.

| propName    | propType | defaultValue | isRequired |
| ----------- | :------: | :----------: | :--------: |
| `className` | `string` |     `''`     |     -      |
| `name`      | `string` |              |     \*     |
| `size`      | `number` |    `3.1`     |     -      |
| `src`       | `string` |    `null`    |     -      |

### Example

```jsx
import Avatar from 'components/Avatar';

render() {
  return <Avatar name={name} src={imageUrl} size={size} />;
}
```
