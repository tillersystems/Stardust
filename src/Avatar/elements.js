import styled from 'styled-components';

export const Image = styled.img`
  border-radius: 50%;
`;
