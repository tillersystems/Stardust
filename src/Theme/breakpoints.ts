/**
 * Defines application breakpoints.
 */

const xs = 0;
const sm = 576;
const md = 768;
const lg = 992;
const xl = 1200;

export const breakpoints = {
  xs,
  sm,
  md,
  lg,
  xl,
};
