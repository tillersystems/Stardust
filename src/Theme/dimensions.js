/**
 * Defines the theming dimensions.
 */

const radius = 0.4;
const tiny = 0.3;
const small = 0.7;
const medium = 1.5;
const big = 3.1;
const huge = 4.5;

export const dimensions = {
  radius: `${radius}rem`,

  tiny: `${tiny}rem`,
  small: `${small}rem`,
  medium: `${medium}rem`,
  big: `${big}rem`,
  huge: `${huge}rem`,

  radiusInt: radius,

  tinyInt: tiny,
  smallInt: small,
  mediumInt: medium,
  bigInt: big,
  hugeInt: huge,
};
