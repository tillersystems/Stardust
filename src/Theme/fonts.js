/**
 * Defines the theming of fonts.
 */

export const fonts = {
  size: {
    // -start- Old font size
    tiny: '0.3rem',
    small: '0.7rem',
    // medium: '1.2rem',
    // big: '3.2rem',
    huge: '6.4rem',
    // -end- Old font size

    // -start- New font size
    default: '1.2rem',
    medium: '1.4rem',
    big: '1.6rem',
    h6: '1.8rem',
    h5: '2.0rem',
    h4: '2.2rem',
    h3: '2.6rem',
    h2: '3rem',
    h1: '3.6rem',
    // -end- New font size

    // -start- Old font int
    tinyInt: 0.3,
    smallInt: 0.7,
    // mediumInt: 1.2,
    // bigInt: 3.2,
    hugeInt: 6.4,
    // -end- Old font int

    // -start- Old font int
    defaultInt: 1.2,
    mediumInt: 1.4,
    bigInt: 1.6,
    h6Int: 1.8,
    h5Int: 2.0,
    h4Int: 2.2,
    h3Int: 2.6,
    h2Int: 3,
    h1Int: 3.6,
    // -end- Old font int
  },

  spacing: {
    small: '0.5px',
    medium: '1px',
    big: '1.5px',
  },

  weight: {
    thin: 200,
    normal: 400,
    thick: 600,
  },

  family: '"Open Sans", Helvetica, Arial, sans-serif',
};
