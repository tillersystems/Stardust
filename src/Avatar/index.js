import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import initials from 'initials';

/**
 * Avatar
 *
 * This component is in charge of displaying
 * an avatar of a user
 *
 * @return {jsx}
 */

import Theme from '../Theme';
import { Image } from './elements';

const defaultColors = [
  Theme.palette.emerald,
  Theme.palette.river,
  Theme.palette.wisteria,
  Theme.palette.carrot,
  Theme.palette.alizarin,
  Theme.palette.turquoise,
];

function sumChars(str) {
  let sum = 0;
  for (let i = 0; i < str.length; i++) {
    sum += str.charCodeAt(i);
  }

  return sum;
}

class Avatar extends PureComponent {
  /**
   * PropTypes validation
   */
  static propTypes = {
    className: PropTypes.string, // className needed by styled component.
    name: PropTypes.string.isRequired, // user name.
    size: PropTypes.number, // size of the avatar.
    src: PropTypes.string, // source image of a user.
  };

  /**
   * Default propTypes
   */
  static defaultProps = {
    className: '',
    size: Theme.dimensions.bigInt,
    src: null,
  };

  /**
   * Render function
   */
  render() {
    const { className, name, size, src } = this.props;
    const abbr = initials(name);
    const normalizeSize = size * 10;

    return (
      <div aria-label={name} className={className}>
        {src && <Image src={src} width={normalizeSize} height={normalizeSize} alt={name} />}
        {!src && <span>{abbr}</span>}
      </div>
    );
  }
}

export default styled(Avatar).attrs({
  colors: ({ name }) => defaultColors[sumChars(name) % defaultColors.length],
})`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${({ size }) => size || Theme.dimensions.bigInt}rem;
  height: ${({ size }) => size || Theme.dimensions.bigInt}rem;
  border-radius: 50%;
  background: ${({ colors }) => colors};
  color: white;
`;
