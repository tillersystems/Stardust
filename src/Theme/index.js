/**
 * Theme exported to the provider
 * This is globally available in styled-components when interpolating a function like so:
 * ${(props) => props.theme}
 * Or using import { withTheme } from 'styled-components';
 */

import { breakpoints } from './breakpoints';
import { palette } from './palette';
import { dimensions } from './dimensions';
import { fonts } from './fonts';

const Theme = {
  breakpoints,
  palette,
  dimensions,
  fonts,
};

export default Theme;
