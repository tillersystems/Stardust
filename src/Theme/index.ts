import { DefaultTheme } from 'styled-components';

/**
 * Theme exported to the provider
 * This is globally available in styled-components when interpolating a function like so:
 * ${(props) => props.theme}
 * Or using import { withTheme } from 'styled-components';
 */

import { breakpoints } from './breakpoints';
import { palette } from './palette';
import { dimensions } from './dimensions';
import { fonts } from './fonts';

const Theme: DefaultTheme = {
  breakpoints,
  palette,
  dimensions,
  fonts,
};

export default Theme;
