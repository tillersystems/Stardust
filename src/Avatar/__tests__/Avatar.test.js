import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import Avatar from '..';
import Theme from '../../Theme';

describe('<Avatar />', () => {
  it('should render withouth a problem', () => {
    const render = mount(<Avatar theme={Theme} name="James Bond" />);
    expect(render).toMatchSnapshot();
  });

  it('should render with a custom user image', () => {
    const render = mount(<Avatar theme={Theme} name="James Bond" src="http://urlofmyimage.com" />);
    expect(render).toMatchSnapshot();
  });

  it('should render withouth a custom size', () => {
    const render = mount(<Avatar theme={Theme} name="James Bond" size={4.1} />);
    expect(render).toMatchSnapshot();
  });
});
