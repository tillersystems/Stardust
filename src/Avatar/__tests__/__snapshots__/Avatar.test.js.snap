// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render with a custom user image 1`] = `
.c1 {
  border-radius: 50%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 3.1rem;
  height: 3.1rem;
  border-radius: 50%;
  background: hsl(28,80%,52%);
  color: white;
}

<Styled(Avatar)
  name="James Bond"
  src="http://urlofmyimage.com"
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "3.2rem",
          "bigInt": 3.2,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.2rem",
          "mediumInt": 1.2,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "alizarin": "hsl(6, 78%, 57%)",
        "anthracite": "hsl(213, 17%, 20%)",
        "black": "hsl(0, 0%, 0%)",
        "blackLight": "hsl(213, 17%, 20%)",
        "blue": "hsl(200, 74%, 46%)",
        "bodyBackground": "hsl(218, 29%, 95%)",
        "carrot": "hsl(28, 80%, 52%)",
        "clay": "hsl(215, 17%, 26%)",
        "darkBlue": "#1c88bf",
        "darkGray": "#dbdfe6",
        "darkGreen": "#7fbf3b",
        "darkRed": "#ea6151",
        "emerald": "hsl(145, 63%, 49%)",
        "fadeBlue": "hsl(210, 50%, 98%)",
        "failure": Object {
          "back": "hsl(6, 79%, 65%)",
          "darkBack": "#ea6151",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#f1978d",
          "lightFore": "#fff",
        },
        "googleBrandRed": "#ea4335",
        "gray": "hsl(218, 18%, 91%)",
        "green": "hsl(89, 53%, 52%)",
        "grey": "hsl(206, 16%, 60%)",
        "greygoose": "hsl(216, 29%, 97%)",
        "lightBlue": "#3baae2",
        "lightGray": "#fff",
        "lightGreen": "#a0d16b",
        "lightRed": "#f1978d",
        "marble": "hsl(206, 23%, 69%)",
        "orange": "hsl(34, 91%, 65%)",
        "primary": Object {
          "back": "hsl(200, 74%, 46%)",
          "darkBack": "#1c88bf",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#3baae2",
          "lightFore": "#fff",
        },
        "red": "hsl(6, 79%, 65%)",
        "river": "hsl(204, 70%, 53%)",
        "sand": "hsl(214, 8%, 84%)",
        "secondary": Object {
          "back": "hsl(0, 100%, 100%)",
          "darkBack": "#f7f7f7",
          "darkFore": "#8093a3",
          "fore": "hsl(206, 16%, 60%)",
          "lightBack": "#fff",
          "lightFore": "#a6b4be",
        },
        "success": Object {
          "back": "hsl(89, 53%, 52%)",
          "darkBack": "#7fbf3b",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#a0d16b",
          "lightFore": "#fff",
        },
        "turquoise": "hsl(168, 76%, 42%)",
        "white": "hsl(0, 100%, 100%)",
        "wisteria": "hsl(282, 44%, 47%)",
      },
    }
  }
>
  <Avatar
    className="c0"
    colors="hsl(28, 80%, 52%)"
    name="James Bond"
    size={3.1}
    src="http://urlofmyimage.com"
    theme={
      Object {
        "breakpoints": Object {
          "lg": 992,
          "md": 768,
          "sm": 576,
          "xl": 1200,
          "xs": 0,
        },
        "dimensions": Object {
          "big": "3.1rem",
          "bigInt": 3.1,
          "huge": "4.5rem",
          "hugeInt": 4.5,
          "medium": "1.5rem",
          "mediumInt": 1.5,
          "radius": "0.4rem",
          "radiusInt": 0.4,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "fonts": Object {
          "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
          "size": Object {
            "big": "3.2rem",
            "bigInt": 3.2,
            "huge": "6.4rem",
            "hugeInt": 6.4,
            "medium": "1.2rem",
            "mediumInt": 1.2,
            "small": "0.7rem",
            "smallInt": 0.7,
            "tiny": "0.3rem",
            "tinyInt": 0.3,
          },
          "spacing": Object {
            "big": "1.5px",
            "medium": "1px",
            "small": "0.5px",
          },
          "weight": Object {
            "normal": 400,
            "thick": 600,
            "thin": 200,
          },
        },
        "palette": Object {
          "alizarin": "hsl(6, 78%, 57%)",
          "anthracite": "hsl(213, 17%, 20%)",
          "black": "hsl(0, 0%, 0%)",
          "blackLight": "hsl(213, 17%, 20%)",
          "blue": "hsl(200, 74%, 46%)",
          "bodyBackground": "hsl(218, 29%, 95%)",
          "carrot": "hsl(28, 80%, 52%)",
          "clay": "hsl(215, 17%, 26%)",
          "darkBlue": "#1c88bf",
          "darkGray": "#dbdfe6",
          "darkGreen": "#7fbf3b",
          "darkRed": "#ea6151",
          "emerald": "hsl(145, 63%, 49%)",
          "fadeBlue": "hsl(210, 50%, 98%)",
          "failure": Object {
            "back": "hsl(6, 79%, 65%)",
            "darkBack": "#ea6151",
            "darkFore": "#f7f7f7",
            "fore": "hsl(0, 100%, 100%)",
            "lightBack": "#f1978d",
            "lightFore": "#fff",
          },
          "googleBrandRed": "#ea4335",
          "gray": "hsl(218, 18%, 91%)",
          "green": "hsl(89, 53%, 52%)",
          "grey": "hsl(206, 16%, 60%)",
          "greygoose": "hsl(216, 29%, 97%)",
          "lightBlue": "#3baae2",
          "lightGray": "#fff",
          "lightGreen": "#a0d16b",
          "lightRed": "#f1978d",
          "marble": "hsl(206, 23%, 69%)",
          "orange": "hsl(34, 91%, 65%)",
          "primary": Object {
            "back": "hsl(200, 74%, 46%)",
            "darkBack": "#1c88bf",
            "darkFore": "#f7f7f7",
            "fore": "hsl(0, 100%, 100%)",
            "lightBack": "#3baae2",
            "lightFore": "#fff",
          },
          "red": "hsl(6, 79%, 65%)",
          "river": "hsl(204, 70%, 53%)",
          "sand": "hsl(214, 8%, 84%)",
          "secondary": Object {
            "back": "hsl(0, 100%, 100%)",
            "darkBack": "#f7f7f7",
            "darkFore": "#8093a3",
            "fore": "hsl(206, 16%, 60%)",
            "lightBack": "#fff",
            "lightFore": "#a6b4be",
          },
          "success": Object {
            "back": "hsl(89, 53%, 52%)",
            "darkBack": "#7fbf3b",
            "darkFore": "#f7f7f7",
            "fore": "hsl(0, 100%, 100%)",
            "lightBack": "#a0d16b",
            "lightFore": "#fff",
          },
          "turquoise": "hsl(168, 76%, 42%)",
          "white": "hsl(0, 100%, 100%)",
          "wisteria": "hsl(282, 44%, 47%)",
        },
      }
    }
  >
    <div
      aria-label="James Bond"
      className="c0"
    >
      <styled.img
        alt="James Bond"
        height={31}
        src="http://urlofmyimage.com"
        width={31}
      >
        <img
          alt="James Bond"
          className="c1"
          height={31}
          src="http://urlofmyimage.com"
          width={31}
        />
      </styled.img>
    </div>
  </Avatar>
</Styled(Avatar)>
`;

exports[`<Avatar /> should render withouth a custom size 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 4.1rem;
  height: 4.1rem;
  border-radius: 50%;
  background: hsl(28,80%,52%);
  color: white;
}

<Styled(Avatar)
  name="James Bond"
  size={4.1}
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "3.2rem",
          "bigInt": 3.2,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.2rem",
          "mediumInt": 1.2,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "alizarin": "hsl(6, 78%, 57%)",
        "anthracite": "hsl(213, 17%, 20%)",
        "black": "hsl(0, 0%, 0%)",
        "blackLight": "hsl(213, 17%, 20%)",
        "blue": "hsl(200, 74%, 46%)",
        "bodyBackground": "hsl(218, 29%, 95%)",
        "carrot": "hsl(28, 80%, 52%)",
        "clay": "hsl(215, 17%, 26%)",
        "darkBlue": "#1c88bf",
        "darkGray": "#dbdfe6",
        "darkGreen": "#7fbf3b",
        "darkRed": "#ea6151",
        "emerald": "hsl(145, 63%, 49%)",
        "fadeBlue": "hsl(210, 50%, 98%)",
        "failure": Object {
          "back": "hsl(6, 79%, 65%)",
          "darkBack": "#ea6151",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#f1978d",
          "lightFore": "#fff",
        },
        "googleBrandRed": "#ea4335",
        "gray": "hsl(218, 18%, 91%)",
        "green": "hsl(89, 53%, 52%)",
        "grey": "hsl(206, 16%, 60%)",
        "greygoose": "hsl(216, 29%, 97%)",
        "lightBlue": "#3baae2",
        "lightGray": "#fff",
        "lightGreen": "#a0d16b",
        "lightRed": "#f1978d",
        "marble": "hsl(206, 23%, 69%)",
        "orange": "hsl(34, 91%, 65%)",
        "primary": Object {
          "back": "hsl(200, 74%, 46%)",
          "darkBack": "#1c88bf",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#3baae2",
          "lightFore": "#fff",
        },
        "red": "hsl(6, 79%, 65%)",
        "river": "hsl(204, 70%, 53%)",
        "sand": "hsl(214, 8%, 84%)",
        "secondary": Object {
          "back": "hsl(0, 100%, 100%)",
          "darkBack": "#f7f7f7",
          "darkFore": "#8093a3",
          "fore": "hsl(206, 16%, 60%)",
          "lightBack": "#fff",
          "lightFore": "#a6b4be",
        },
        "success": Object {
          "back": "hsl(89, 53%, 52%)",
          "darkBack": "#7fbf3b",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#a0d16b",
          "lightFore": "#fff",
        },
        "turquoise": "hsl(168, 76%, 42%)",
        "white": "hsl(0, 100%, 100%)",
        "wisteria": "hsl(282, 44%, 47%)",
      },
    }
  }
>
  <Avatar
    className="c0"
    colors="hsl(28, 80%, 52%)"
    name="James Bond"
    size={4.1}
    src={null}
    theme={
      Object {
        "breakpoints": Object {
          "lg": 992,
          "md": 768,
          "sm": 576,
          "xl": 1200,
          "xs": 0,
        },
        "dimensions": Object {
          "big": "3.1rem",
          "bigInt": 3.1,
          "huge": "4.5rem",
          "hugeInt": 4.5,
          "medium": "1.5rem",
          "mediumInt": 1.5,
          "radius": "0.4rem",
          "radiusInt": 0.4,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "fonts": Object {
          "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
          "size": Object {
            "big": "3.2rem",
            "bigInt": 3.2,
            "huge": "6.4rem",
            "hugeInt": 6.4,
            "medium": "1.2rem",
            "mediumInt": 1.2,
            "small": "0.7rem",
            "smallInt": 0.7,
            "tiny": "0.3rem",
            "tinyInt": 0.3,
          },
          "spacing": Object {
            "big": "1.5px",
            "medium": "1px",
            "small": "0.5px",
          },
          "weight": Object {
            "normal": 400,
            "thick": 600,
            "thin": 200,
          },
        },
        "palette": Object {
          "alizarin": "hsl(6, 78%, 57%)",
          "anthracite": "hsl(213, 17%, 20%)",
          "black": "hsl(0, 0%, 0%)",
          "blackLight": "hsl(213, 17%, 20%)",
          "blue": "hsl(200, 74%, 46%)",
          "bodyBackground": "hsl(218, 29%, 95%)",
          "carrot": "hsl(28, 80%, 52%)",
          "clay": "hsl(215, 17%, 26%)",
          "darkBlue": "#1c88bf",
          "darkGray": "#dbdfe6",
          "darkGreen": "#7fbf3b",
          "darkRed": "#ea6151",
          "emerald": "hsl(145, 63%, 49%)",
          "fadeBlue": "hsl(210, 50%, 98%)",
          "failure": Object {
            "back": "hsl(6, 79%, 65%)",
            "darkBack": "#ea6151",
            "darkFore": "#f7f7f7",
            "fore": "hsl(0, 100%, 100%)",
            "lightBack": "#f1978d",
            "lightFore": "#fff",
          },
          "googleBrandRed": "#ea4335",
          "gray": "hsl(218, 18%, 91%)",
          "green": "hsl(89, 53%, 52%)",
          "grey": "hsl(206, 16%, 60%)",
          "greygoose": "hsl(216, 29%, 97%)",
          "lightBlue": "#3baae2",
          "lightGray": "#fff",
          "lightGreen": "#a0d16b",
          "lightRed": "#f1978d",
          "marble": "hsl(206, 23%, 69%)",
          "orange": "hsl(34, 91%, 65%)",
          "primary": Object {
            "back": "hsl(200, 74%, 46%)",
            "darkBack": "#1c88bf",
            "darkFore": "#f7f7f7",
            "fore": "hsl(0, 100%, 100%)",
            "lightBack": "#3baae2",
            "lightFore": "#fff",
          },
          "red": "hsl(6, 79%, 65%)",
          "river": "hsl(204, 70%, 53%)",
          "sand": "hsl(214, 8%, 84%)",
          "secondary": Object {
            "back": "hsl(0, 100%, 100%)",
            "darkBack": "#f7f7f7",
            "darkFore": "#8093a3",
            "fore": "hsl(206, 16%, 60%)",
            "lightBack": "#fff",
            "lightFore": "#a6b4be",
          },
          "success": Object {
            "back": "hsl(89, 53%, 52%)",
            "darkBack": "#7fbf3b",
            "darkFore": "#f7f7f7",
            "fore": "hsl(0, 100%, 100%)",
            "lightBack": "#a0d16b",
            "lightFore": "#fff",
          },
          "turquoise": "hsl(168, 76%, 42%)",
          "white": "hsl(0, 100%, 100%)",
          "wisteria": "hsl(282, 44%, 47%)",
        },
      }
    }
  >
    <div
      aria-label="James Bond"
      className="c0"
    >
      <span>
        JB
      </span>
    </div>
  </Avatar>
</Styled(Avatar)>
`;

exports[`<Avatar /> should render withouth a problem 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 3.1rem;
  height: 3.1rem;
  border-radius: 50%;
  background: hsl(28,80%,52%);
  color: white;
}

<Styled(Avatar)
  name="James Bond"
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "3.2rem",
          "bigInt": 3.2,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.2rem",
          "mediumInt": 1.2,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "alizarin": "hsl(6, 78%, 57%)",
        "anthracite": "hsl(213, 17%, 20%)",
        "black": "hsl(0, 0%, 0%)",
        "blackLight": "hsl(213, 17%, 20%)",
        "blue": "hsl(200, 74%, 46%)",
        "bodyBackground": "hsl(218, 29%, 95%)",
        "carrot": "hsl(28, 80%, 52%)",
        "clay": "hsl(215, 17%, 26%)",
        "darkBlue": "#1c88bf",
        "darkGray": "#dbdfe6",
        "darkGreen": "#7fbf3b",
        "darkRed": "#ea6151",
        "emerald": "hsl(145, 63%, 49%)",
        "fadeBlue": "hsl(210, 50%, 98%)",
        "failure": Object {
          "back": "hsl(6, 79%, 65%)",
          "darkBack": "#ea6151",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#f1978d",
          "lightFore": "#fff",
        },
        "googleBrandRed": "#ea4335",
        "gray": "hsl(218, 18%, 91%)",
        "green": "hsl(89, 53%, 52%)",
        "grey": "hsl(206, 16%, 60%)",
        "greygoose": "hsl(216, 29%, 97%)",
        "lightBlue": "#3baae2",
        "lightGray": "#fff",
        "lightGreen": "#a0d16b",
        "lightRed": "#f1978d",
        "marble": "hsl(206, 23%, 69%)",
        "orange": "hsl(34, 91%, 65%)",
        "primary": Object {
          "back": "hsl(200, 74%, 46%)",
          "darkBack": "#1c88bf",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#3baae2",
          "lightFore": "#fff",
        },
        "red": "hsl(6, 79%, 65%)",
        "river": "hsl(204, 70%, 53%)",
        "sand": "hsl(214, 8%, 84%)",
        "secondary": Object {
          "back": "hsl(0, 100%, 100%)",
          "darkBack": "#f7f7f7",
          "darkFore": "#8093a3",
          "fore": "hsl(206, 16%, 60%)",
          "lightBack": "#fff",
          "lightFore": "#a6b4be",
        },
        "success": Object {
          "back": "hsl(89, 53%, 52%)",
          "darkBack": "#7fbf3b",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#a0d16b",
          "lightFore": "#fff",
        },
        "turquoise": "hsl(168, 76%, 42%)",
        "white": "hsl(0, 100%, 100%)",
        "wisteria": "hsl(282, 44%, 47%)",
      },
    }
  }
>
  <Avatar
    className="c0"
    colors="hsl(28, 80%, 52%)"
    name="James Bond"
    size={3.1}
    src={null}
    theme={
      Object {
        "breakpoints": Object {
          "lg": 992,
          "md": 768,
          "sm": 576,
          "xl": 1200,
          "xs": 0,
        },
        "dimensions": Object {
          "big": "3.1rem",
          "bigInt": 3.1,
          "huge": "4.5rem",
          "hugeInt": 4.5,
          "medium": "1.5rem",
          "mediumInt": 1.5,
          "radius": "0.4rem",
          "radiusInt": 0.4,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "fonts": Object {
          "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
          "size": Object {
            "big": "3.2rem",
            "bigInt": 3.2,
            "huge": "6.4rem",
            "hugeInt": 6.4,
            "medium": "1.2rem",
            "mediumInt": 1.2,
            "small": "0.7rem",
            "smallInt": 0.7,
            "tiny": "0.3rem",
            "tinyInt": 0.3,
          },
          "spacing": Object {
            "big": "1.5px",
            "medium": "1px",
            "small": "0.5px",
          },
          "weight": Object {
            "normal": 400,
            "thick": 600,
            "thin": 200,
          },
        },
        "palette": Object {
          "alizarin": "hsl(6, 78%, 57%)",
          "anthracite": "hsl(213, 17%, 20%)",
          "black": "hsl(0, 0%, 0%)",
          "blackLight": "hsl(213, 17%, 20%)",
          "blue": "hsl(200, 74%, 46%)",
          "bodyBackground": "hsl(218, 29%, 95%)",
          "carrot": "hsl(28, 80%, 52%)",
          "clay": "hsl(215, 17%, 26%)",
          "darkBlue": "#1c88bf",
          "darkGray": "#dbdfe6",
          "darkGreen": "#7fbf3b",
          "darkRed": "#ea6151",
          "emerald": "hsl(145, 63%, 49%)",
          "fadeBlue": "hsl(210, 50%, 98%)",
          "failure": Object {
            "back": "hsl(6, 79%, 65%)",
            "darkBack": "#ea6151",
            "darkFore": "#f7f7f7",
            "fore": "hsl(0, 100%, 100%)",
            "lightBack": "#f1978d",
            "lightFore": "#fff",
          },
          "googleBrandRed": "#ea4335",
          "gray": "hsl(218, 18%, 91%)",
          "green": "hsl(89, 53%, 52%)",
          "grey": "hsl(206, 16%, 60%)",
          "greygoose": "hsl(216, 29%, 97%)",
          "lightBlue": "#3baae2",
          "lightGray": "#fff",
          "lightGreen": "#a0d16b",
          "lightRed": "#f1978d",
          "marble": "hsl(206, 23%, 69%)",
          "orange": "hsl(34, 91%, 65%)",
          "primary": Object {
            "back": "hsl(200, 74%, 46%)",
            "darkBack": "#1c88bf",
            "darkFore": "#f7f7f7",
            "fore": "hsl(0, 100%, 100%)",
            "lightBack": "#3baae2",
            "lightFore": "#fff",
          },
          "red": "hsl(6, 79%, 65%)",
          "river": "hsl(204, 70%, 53%)",
          "sand": "hsl(214, 8%, 84%)",
          "secondary": Object {
            "back": "hsl(0, 100%, 100%)",
            "darkBack": "#f7f7f7",
            "darkFore": "#8093a3",
            "fore": "hsl(206, 16%, 60%)",
            "lightBack": "#fff",
            "lightFore": "#a6b4be",
          },
          "success": Object {
            "back": "hsl(89, 53%, 52%)",
            "darkBack": "#7fbf3b",
            "darkFore": "#f7f7f7",
            "fore": "hsl(0, 100%, 100%)",
            "lightBack": "#a0d16b",
            "lightFore": "#fff",
          },
          "turquoise": "hsl(168, 76%, 42%)",
          "white": "hsl(0, 100%, 100%)",
          "wisteria": "hsl(282, 44%, 47%)",
        },
      }
    }
  >
    <div
      aria-label="James Bond"
      className="c0"
    >
      <span>
        JB
      </span>
    </div>
  </Avatar>
</Styled(Avatar)>
`;
